### PR TITLE
Integrate output search api

### LINKED ISSUE #1828: Revisit HTTP API for tendermint integration
- [x] Identify tendermint endpoints and their purpose
- [x] Bind bigchaindb endpoints to tendermint ones, when relevant.

### PR DESCRIPTION
**Pull request by kansi**
## Description
Integrate output search api to tendermint implementation.

## Issues This PR Fixes
Fixes #1828

### CODE REVIEW

File: tests/web/test_outputs.py [hunk 1]
```diff
@@ -6,11 +6,16 @@
 OUTPUTS_ENDPOINT = '/api/v1/outputs/'
 
 
-def test_get_outputs_endpoint(client, user_pk):
+@pytest.mark.tendermint
+@pytest.mark.parametrize("bigchain_class", [
+    'bigchaindb.core.Bigchain.get_outputs_filtered',
```
Comment (vrde) [inline tests/web/test_outputs.py]: We can drop testing `bigchaindb.core.Bigchain.get_outputs_filtered`.

File: tests/web/conftest.py [hunk 2]
```diff
@@ -3,11 +3,10 @@
 
 @pytest.fixture
 def app(request):
-    from bigchaindb import config
     from bigchaindb.web import server
     from bigchaindb.tendermint.lib import BigchainDB
 
-    if config['database']['backend'] == 'localmongodb':
+    if request.config.getoption('--database-backend') == 'localmongodb':
```
[the remark this answers is not in the document]
Comment (kansi) [reply tests/web/conftest.py]: I am doing this becuase the test suite doesn't pick up the config i.e. I generate the config for `localmongodb` but the test suite still loads for `rethinkdb`.

File: tests/web/test_outputs.py (hunk 1 above)
Comment (kansi) [reply tests/web/test_outputs.py]: We could drop it if we are only testing for tendermint but if we use `mongodb` as backned then we need to test `bigchaindb.core.Bigchain.get_outputs_filtered` as the web view uses this function to prepare the response.

File: tests/backend/localmongodb/test_queries.py [hunk 3]
```diff
@@ -74,8 +71,40 @@ def test_get_assets():
...
 
 
-@pytest.mark.bdb
 def test_text_search():
     from ..mongodb.test_queries import test_text_search
 
     test_text_search()
+
+
+def test_get_owned_ids(signed_create_tx, user_pk):
```
[the remark this answers is not in the document]
Comment (kansi) [reply tests/backend/localmongodb/test_queries.py]: [Yes](https://github.com/bigchaindb/bigchaindb/blob/feat/integrate-output-api/tests/backend/mongodb/test_queries.py#L203), but the test can't be directly ported.

File: bigchaindb/tendermint/fastquery.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+from bigchaindb.utils import condition_details_has_owner
+from bigchaindb.backend import query
+from bigchaindb.common.transaction import TransactionLink
+
+
+class FastQuery(object):
```
Comment (vrde) [inline bigchaindb/tendermint/fastquery.py]: In Python 3 `(object)` is no longer needed.

**Pull request merged**